tiledb/bioimg/converters/axes.py

```diff
@@ -87,19 +85,19 @@ def transpose_array(a: np.ndarray, s: Sequence[T], t: Sequence[T]) -> np.ndarray
     return a
 
 
-def minimize_transpositions(s: Sequence[T], t: Sequence[T]) -> Sequence[Transpose]:
+def minimize_transpositions(s: str, t: str) -> Sequence[Transpose]:
     assert Counter(s) == Counter(t)
     n = len(s)
-    s = list(s)
-    t = list(t)
+    sbuf = bytearray(s.encode())
+    tbuf = t.encode()
     transpositions = []
-    while s != t:
+    while sbuf != tbuf:
         weighted_transpositions = (
-            (Levenshtein.distance(transposition.transposed(s), t), transposition)
-            for transposition in gen_transpositions(n)
+            (levenshtein.distance(tr.transposed(sbuf).decode(), t), tr)
```

Comment:
Any gain in case we parallelize here?

Reply:
Haven't checked but it's not worth it, the compared strings are tiny (2-5 chars long) and `levenshtein.distance` is in C++/Cython so the whole `minimize_transpositions` call takes less than a msec.